pkg/runtime/runtime.go

```diff
@@ -1323,9 +1324,24 @@ func (a *DaprRuntime) processComponents() {
 		if comp.Name == "" {
 			continue
 		}
-		err := a.processComponentAndDependents(comp)
+
+		ch := make(chan error, 1)
+		go func() {
+			ch <- a.processComponentAndDependents(comp)
+		}()
+
+		timeout, err := time.ParseDuration(comp.Spec.InitTimeout)
 		if err != nil {
-			log.Errorf("process component %s error, %s", comp.Name, err)
+			timeout = defaultComponentInitTimeout
+		}
+
+		select {
+		case err := <-ch:
+			if err != nil {
+				log.Errorf("process component %s error, %s", comp.Name, err)
+			}
+		case <-time.After(timeout):
+			log.Errorf("init timeout for component %s exceeded after %s", comp.Name, timeout.String())
```

Comment:
Ideally, we should cancel the operation. But goroutine does not support that.

Reply:
Changed

Reply:
We could pass the context down to callees, so they can, in turn, pass along to any blocking / long-running operations (like `http.Client.Do()`, or do `select ... <-ctx.Done()` themselves.

Reply:
That's the long term plan, we should have an issue for this. Not all IO packages accept a context though